src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/MovementService.kt

```diff
@@ -1,12 +1,59 @@
 package uk.gov.justice.digital.hmpps.welcometoprison.model
 
+import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.welcometoprison.model.MoveType.PRISON_TRANSFER
 import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.BasmService
 import uk.gov.justice.digital.hmpps.welcometoprison.model.prison.PrisonService
+import uk.gov.justice.digital.hmpps.welcometoprison.model.prisonersearch.PrisonerSearchService
+import uk.gov.justice.digital.hmpps.welcometoprison.model.prisonersearch.response.MatchPrisonerResponse
 import java.time.LocalDate
 
 @Service
-class MovementService(val basmService: BasmService, val prisonService: PrisonService) {
+class MovementService(
+  val basmService: BasmService,
+  val prisonService: PrisonService,
+  val prisonerSearchService: PrisonerSearchService
+) {
   fun getMovements(agencyId: String, date: LocalDate) =
     basmService.getMoves(agencyId, date, date) + prisonService.getMoves(agencyId, date)
+
+  fun getMovementsMatchedWithPrisoner(agencyId: String, date: LocalDate): List<Movement> {
+    val movements = getMovements(agencyId, date)
+    return getMovementsMatchedWithPrisoner(movements)
+  }
+
+  private fun getMovementsMatchedWithPrisoner(movements: List<Movement>): List<Movement> {
+    fun findPrisonerMatch(identifier: String?, identifierName: String): MatchPrisonerResponse? {
+      val matchesByPrisonNumber = identifier?.let { prisonerSearchService.matchPrisoner(it) } ?: emptyList()
+      if (matchesByPrisonNumber.size > 1) log.warn("Multiple matched Prison records for a Movement by $identifierName. There are ${matchesByPrisonNumber.size} matched Prison records for $identifier")
+      return matchesByPrisonNumber.firstOrNull()
+    }
+    fun decorateMovementWithPrisonerMatch(movement: Movement): Movement {
+      var prisonNumberToUse = movement.prisonNumber
+      var pncToUse = movement.pncNumber
+
+      val matchByPrisonNumber = findPrisonerMatch(movement.prisonNumber, "Prison Number")
+      val matchByPncNumber = findPrisonerMatch(movement.pncNumber, "PNC Number")
```

Comment:
by doing this in a loop it means there will be n x 2 calls and none of these are parallelised.. this could be a bit slow. 
Maybe that will do fine for now but we may need to optimise this.  

There is an endpoint for retrieving [a list of prisoners by prison number](https://prisoner-offender-search-dev.prison.service.justice.gov.uk/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/prisoner-search-resource/findByNumbers) 

I think I added one for bookings IDs also and probably not too hard to add one for PNC

Reply:
Added a ticket https://dsdmoj.atlassian.net/browse/DCS-1222